### DISCUSSION TITLE
Why can't I get ads to appear consistently?

### DISCUSSION THREAD

**Discussion by Jordancjb** [Poing-Studios/Godot-AdMob-Android-iOS#56] [Q&A] [answered]
I feel this is my fault but using godot 3.3 I can't get any ads to appear consistently(or sometimes at all)

Banner ads running load_banner() only seem to work if I make a timer of 1-2 seconds off my _ready() func

Interstitial ads don't even appear for me

I can't use any of the signals if I do MobileAds.connect()

It is a singleton/autoload

It just doesn't seem consistent and though It seems very nice compared to my other option(s) It just doesn't seem reliable

If I am missing something please let me know

**Comment by gumaciel** [comment]
Hello, what OS are you using to test? Android or iOS?

Are you trying to use the AdMob with the example project? https://github.com/Poing-Studios/Godot-AdMob-Android-iOS/tree/master/example

**Reply by Jordancjb** [reply]
I'm just testing on an android tablet

And I am just trying to add admob to a mobile game I'm making for google play

**Reply by gumaciel** [reply]
@Jordancjb please test this apk and also test with the example project, if still doesn't work, can you record a video and send here to see what the app is doing? Thanks

https://github.com/Poing-Studios/Godot-AdMob-Android-iOS/discussions/56#discussioncomment-667457

**Reply by Jordancjb** [reply]
my device won't let me? I downloaded the apk but I try and my device won't download it despite me accepting every warning It gives me

**Reply by gumaciel** [reply]
You were probably needing to activate the developer mode on your phone, it is normal to give these warnings because installing internet applications in this way can be dangerous for your phone and may contain a virus or something.

**Comment by Jordancjb** [comment, marked as answer]
Well nvm my issue is fixed

I was just missing a step and being dumb
This plugin works amazing!

**Reply by gumaciel** [reply]
Excellent! I'm still improving the instructions for use and installation, any questions just ask again.

**Comment by frederickiv43** [comment]
Hello, I am having the same issue as described in the OP. Testing on Android with Godot 3.4. Banner ads only working after a timer, Interstitial ads not showing at all. Any suggestions?

**Reply by gumaciel** [reply]
many people are complaining about this issue, and there was no change of source code to load or display ads, what i think is that google changed some of the demo ad ids, please check here: 

https://developers.google.com/admob/ios/test-ads#demo_ad_units
https://developers.google.com/admob/android/test-ads#demo_ad_units

it could also be your internet connection, try using a VPN

if it doesn't work, try creating an Id Unit Ad on admob.com, remember that it may take a while to work, read more here: https://support.google.com/admob/answer/9469204?hl=en
